### ISSUE TITLE
[File-based-router] Layout nested in subfolder

### ISSUE THREAD

**Issue by simonedelmann** [TanStack/router#1174]
### Describe the bug

Discussed in Discord... 

The problem:
If a layout file (`_layout.tsx`) is in a subfolder (`routes/some-subfolder/_layout.tsx`), the layout will only get removed from the url if the subfolder is a route node (= if either `routes/some-subfolder.tsx` or `routes/some-subfolder/route.tsx` exists). 

The reason for this seems to be that TSR sees that `some-subfolder` is no route itself, so it just prefixes the path of it's children with `/some-subfolder`, which causes the prefix underscore `_` to be ignored. 

### Your Example Website or App

https://stackblitz.com/edit/vitejs-vite-8xg9aw?file=src%2Froutes%2F__root.tsx

### Steps to Reproduce the Bug or Issue

File `routes/employee/_features/cart/index.tsx` gets generated as `/employee/features/cart`.

### Expected behavior

File `routes/employee/_features/cart/index.tsx` should have path `/employee/cart`.

### Platform

...

**Comment by SeanCassiere** [comment]
This looks to be another instance of the bug reported in #1150.

**Comment by simonedelmann** [comment]
You are right! Sorry, I missed that. Closing this for https://github.com/TanStack/router/issues/1150 .
